scripts/validate-artifacts.ps1

```diff
@@ -34,142 +34,155 @@ Write-Output "Validating artifacts"
-    if ($virtualMachine.properties.availabilitySet.id) { 
-        Write-Output "`u{2705} Checked if virtual machine has availability set assigned - OK"
-    } else  {
-        Write-Output `u{1F914}
-        throw "Unable to verify that VM is assigned to the availability set. Please make sure that you are assinging the VM to the availability set during the creation with parameter '-AvailabilitySetName' and try again."
-    }
 
-    if ($virtualMachine.properties.osProfile.linuxConfiguration.ssh.publicKeys.keyData -eq $sshKey.properties.publicKey) { 
-        Write-Output "`u{2705} Checked if virtual machine uses the public ssh key 'linuxboxsshkey' - OK"
-    } else { 
+$nsg = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups")
+if ($nsg) {
+    if ($nsg.name.Count -eq 1) { 
+        Write-Output "`u{2705} Checked if the Network Security Group resource exists - OK"
+    }  else { 
         Write-Output `u{1F914}
-        throw "Unable to verify that VM uses the public ssh key 'linuxboxsshkey'. Please make sure that in New-AzVm comandled, parameter '-SshKeyName' is set to the name of the public SSH key you created earlier, and that you are not setting the parameter '-GenerateSshKey'."
+        throw "More than one Network Security Group resource was found in the VM resource group. Please delete all un-used Network Security Group resources and try again."
     }
+} else {
+    Write-Output `u{1F914}
+    throw "Unable to find Network Security Group resouce. Please re-deploy the VM and try again."
+}
 
-    if ($virtualMachine.properties.storageProfile.imageReference.publisher -eq "canonical") { 
-        Write-Output "`u{2705} Checked Virtual Machine OS image publisher - OK" 
-    } else { 
-        Write-Output `u{1F914}
-        throw "Virtual Machine uses OS image from unknown published. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again."
-    }
-    if ($virtualMachine.properties.storageProfile.imageReference.offer.Contains('ubuntu-server') -and $virtualMachine.properties.storageProfile.imageReference.sku.Contains('22_04')) { 
-        Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
-    } else { 
-        Write-Output `u{1F914}
-        throw "Virtual Machine uses wrong OS image. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again." 
-    }
+$sshNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '22') -and ($_.properties.access -eq 'Allow')} ) 
+if ($sshNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has SSH network security rule configured - OK"
+} else { 
+    Write-Output `u{1F914}
+    throw "Unable to fing network security group rule which allows SSH connection. Please check if you configured VM Network Security Group to allow connections on 22 TCP port and try again."
```

Comment:
There is a typo in the error message: 'fing' should be 'find'. This is a minor issue but could cause confusion.